### DISCUSSION TITLE
I only see 12h/13h data on grafana

### DISCUSSION THREAD

**Discussion by dadoMT** [grafana/mimir#7499] [General]
Hi everyone,

can anyone explain why I can't see old data when I do a query from grafana?

![image](https://github.com/grafana/mimir/assets/142960252/48e2aa4e-ed76-4346-8c03-45ac6fd5aac7)

I'm using the mimir-distributed helm chart and have NAS as long-term object storage

**Comment by dadoMT** [comment]
now, I just refreshed and looked at the mimir-mixin dashboards and the ingester is unhealthy again. how do I fix this? :(

![image](https://github.com/grafana/mimir/assets/142960252/ec52b716-6c58-4dc3-8f3c-96564f0a5376)

Loki logs:

errors in sequence 6h:

{"log":"ts=2024-02-28T22:05:49.070350109Z caller=ingester.go:2554 level=error msg=\"error while opening existing TSDBs\" err=\"unable to open TSDB for user default_org: failed to open TSDB: /nas_path/mimir/tsdb/default_org: get segment range: segments are not sequential\"\n","stream":"stderr","time":"2024-02-28T22:05:49.092762289Z"}

{"log":"ts=2024-02-28T22:05:49.169287567Z caller=mimir.go:869 level=error msg=\"module failed\" module=ingester-service err=\"starting module ingester-service: invalid service state: Failed, expected: Running, failure: opening existing TSDBs: unable to open TSDB for user default_org: failed to open TSDB: /nas_path/mimir/tsdb/default_org: get segment range: segments are not sequential\"\n","stream":"stderr","time":"2024-02-28T22:05:49.192889917Z"}

{"log":"ts=2024-02-28T22:05:49.718880351Z caller=log.go:103 level=error msg=\"error running application\" err=\"failed services\\ngithub.com/grafana/mimir/pkg/mimir.(*Mimir).Run\\n\\t/__w/mimir/mimir/pkg/mimir/mimir.go:912\\nmain.main\\n\\t/__w/mimir/mimir/cmd/mimir/main.go:227\\nruntime.main\\n\\t/usr/local/go/src/runtime/proc.go:267\\nruntime.goexit\\n\\t/usr/local/go/src/runtime/asm_amd64.s:1650\"\n","stream":"stderr","time":"2024-02-28T22:05:49.719067516Z"}

how do I fix please?

**Reply by dadoMT** [reply]
well I fixed the ingester issue by moving the corrupted block. but I want to figure out how it got corrupted in the first place? any ideas please? and can I fix the corrupted data? I have it in my backup folder

aside from that. why is that I only see 12/13h of data?

![image](https://github.com/grafana/mimir/assets/142960252/aefce26b-dd88-49c3-adbb-7fba7b6f4b48)

there's a lot from yesterday that's not showing which is in the long-term storage

![image](https://github.com/grafana/mimir/assets/142960252/090115f2-f289-4e97-9d16-a3d84773d187)

**Comment by dadoMT** [comment]
not sure why no one replied on this discussion, to fix this one. we just moved to a private s3 storage (dell ecs) instead of using NAS.

This fixed the corrupted data issue, as well as the missing data past 12/13h. 

Please update your documentation regarding Object Storage (especially filesystem setup)

on the Prometheus Storage Documentation this is included:

`CAUTION: Non-POSIX compliant filesystems are not supported for Prometheus' local storage as unrecoverable corruptions may happen. NFS filesystems (including AWS's EFS) are not supported. NFS could be POSIX-compliant, but most implementations are not. It is strongly recommended to use a local filesystem for reliability.`

reference: https://prometheus.io/docs/prometheus/latest/storage/

I only see on the logs to use s3 / cloud storage. There's not much reason why you shouldn't use NFS or NAS or any other file systems for production use

It also say to use distributed filesystem from the logs (which we believe is what NAS is). please explain that properly on your documentation

**Comment by lianghuiyuan** [comment]
I get the same problem. 
I know the time is controled by `query_ingesters_within: 13h` and  `query_store_after: 12h`, 
but I have no idea why grafana can't get more data from mimir than 13 hours ago.

**Comment by vsile** [comment]
The same issue with Mimir 2.16.1.
We have a flag --blocks-storage.tsdb.retention-period=720h. 
We can see the blocks (for 720 hours) via Mimir UI (Ingester Page).
<img width="1553" height="1040" alt="image" src="https://github.com/user-attachments/assets/0719b72d-9b5c-4c6c-9c25-4a724879f695" />

But Grafana shows the data only for 13 hours..

<img width="931" height="260" alt="Screenshot_20250813_011253" src="https://github.com/user-attachments/assets/910c097f-53f8-4967-9061-e63741972566" />

**Reply by vsile** [reply]
Well, 
we set the flag **--querier.query-ingesters-within=0** and the issue has been fixed.
<img width="936" height="264" alt="Screenshot_20250813_021638" src="https://github.com/user-attachments/assets/9f04d090-616d-4883-890c-cccd8de0b394" />
